utils/src/AzExtTreeFileSystem.ts

```diff
@@ -8,25 +8,20 @@ import { Disposable, Event, EventEmitter, FileChangeEvent, FileStat, FileSystemE
 import * as types from '../index';
 import { callWithTelemetryAndErrorHandling } from "./callWithTelemetryAndErrorHandling";
 import { localize } from "./localize";
-import { AzExtTreeDataProvider } from "./tree/AzExtTreeDataProvider";
-import { AzExtTreeItem } from "./tree/AzExtTreeItem";
 import { nonNullProp } from "./utils/nonNull";
 
 const unsupportedError: Error = new Error(localize('notSupported', 'This operation is not supported.'));
 
-export abstract class AzExtTreeFileSystem<TItem extends AzExtTreeItem> implements FileSystemProvider {
+export abstract class AzExtTreeFileSystem<TItem extends types.AzExtTreeFileSystemItem> implements FileSystemProvider {
+
+    private readonly itemCache: Map<string, TItem> = new Map<string, TItem>();
```

Comment:
Should we clear and repopulate this every time a refresh happens? Seems like it could cause stale tree item ids if we're not careful.

Reply:
I haven't run into that in my testing, but I am worried about the possibility of something like that happening. The tree item ids correspond to Azure resource ids so unless something is deleted I think it's fine. I will test that now.

Reply:
Also whenever I tried clearing this map, it resulted in errors since we can't really know if the items in the map are still in use easily.

Reply:
A potential fix for all of this would be to really remove the reliance on items at all, and base if fully on URIs.